bl.c

```diff
@@ -328,16 +329,19 @@ jump_to_app()
 	vec_base = (const uint32_t *)0;
 
 	/* TOC not found or empty, stay in btl */
-	if (!find_toc(&toc_entries, &len)) {
+	if (!find_toc(&toc_start, &len)) {
 		return;
 	}
 
 	/* Verify the first entry, containing the TOC itself */
-	if (!verify_app(0, toc_entries)) {
+	if (!verify_toc(toc_start)) {
```

Comment:
This is the part which I referred to be too complicated, and does not work.

This doesn't work in case TOC is embedded in the first application. toc_start points to the beginning of the TOC (in the middle of the first entry), and the verification should always be done over the first entry, no matter whether it is TOC only or something else in addition.

Reply:
Sorry, I think I misunderstood this. It still takes the first entry from the TOC. It was just confusing and I didn't get it.

Reply:
I guess it is fine if it is tested to work :)

Reply:
Sorry for going back and forth; I think the original line is cleaner, can you preserve that? It tells exactly what it does, without need to dig into several files to find out. It verifies the entry 0, and it could stand like that in the code already in here. The comment there tells the story correctly, the first entry contains the TOC, so it gets verified here.

Reply:
Sure, I'll change it. The procedure "verify_toc" is really just verify_app for the first entry anyway.